### DISCUSSION TITLE
2021-10-05 weekly planning

### DISCUSSION THREAD

**Discussion by kgodey** [mathesar-foundation/mathesar#693] [Maintainer discussions]
### Discussion topics

- **[Updates](https://github.com/centerofci/mathesar/discussions/693#discussioncomment-1432330)**: General updates everyone should be aware of.
- **[Questions](https://github.com/centerofci/mathesar/discussions/693#discussioncomment-1432332)**: Questions for the team to answer.
- **[Comms Liaison Schedule](https://github.com/centerofci/mathesar/discussions/693#discussioncomment-1432334)**: Upcoming comms liaison schedule for the core team.
- **[Active Work](https://github.com/centerofci/mathesar/discussions/693#discussioncomment-1432337)**: Work planned for the next week.
- **[Retrospective](https://github.com/centerofci/mathesar/discussions/693#discussioncomment-1432345)**: Summary of last week's work.

### About
Each Tuesday, we start a discussion about Mathesar's progress. Everyone is welcome to add a new topic to discuss or to comment on existing topics.

**Comment by kgodey** [comment]
## Updates

### Staging Server

We finally have a permanent Mathesar server up: https://staging.mathesar.org/. The latest Mathesar code is automatically deployed to it whenever a new pull request is merged.

The server is currently password-protected because we don't have any authentication in the product yet. Please talk to me if you want access.

### Progress

We're continuing to make progress on our active milestones:
- Frontend work on [[05] Editable Tables](https://github.com/centerofci/mathesar/milestone/4) is almost complete. A design review of the "adding records" functionality last week resulted in some major design changes, but once those are done, we can close the milestone.
- Frontend, backend, and design work on the October stability release, [[06] 2021-10 Stability](https://github.com/centerofci/mathesar/milestone/34) is ongoing.
- Some frontend work on [[07] Working with Tables](https://github.com/centerofci/mathesar/milestone/8) by community contributors is in review and should be merged soon. All backend work has been completed.
- Backend work on [[08] Initial Data Types](https://github.com/centerofci/mathesar/milestone/11) is in progress. It's a little slower than anticipated because of issues with the functionality involved in setting default values, but those are in review now. We're making steady progress, though, with text types completed and URI types in review.
- Design work on [[09] Working with Views](https://github.com/centerofci/mathesar/milestone/13) is progressing nicely and will hopefully be wrapped up in the next week.

### Hiring

Interviews for the [backend engineer](https://wiki.mathesar.org/jobs) position are in full swing. We're still accepting applications and referrals are appreciated.

### Design

We now have a new repository: https://github.com/centerofci/mathesar-design/ for HTML/CSS design prototypes. For features involving a lot of user interactions, working with HTML prototypes was easier than Figma. We will be incorporating these designs into our in-review and future design specs.

You can see the current state of the prototype at https://mathesar-prototype.netlify.app/. It is still a work in progress.

### Community Comms Liaison

We've decided to designate a single person on the core team each work to be on point for community-related communications (Matrix messages, pull request reviews, issue comments, etc.). This person will pull in other people on the team as needed. This arrangement will allow us to respond to community messages in a timely manner without needing everyone on the team to pay attention to messages all the time.

We'll use the weekly discussions to designate the comms liaison each week.

**Comment by kgodey** [comment]
## Questions

@mathemancer, @ghislaineguerin, @pavish: Your responses are encouraged. Others are also welcome to respond.

- What went well over the past week?
- What could have gone better over the past week?
- Does your workload for this week look okay?
- Are there any improvements you'd like to suggest?

**Reply by ghislaineguerin** [reply]
> What went well over the past week?

I've been focusing on design problems for longer periods of time. Also, coordinating w Kriti to send the project list for students.

> What could have gone better over the past week?

I still don't have a plan for delivering the UI styling.
I also wish I could contribute more to discussions on our design channel and do more community-facing activities.

> Does your workload for this week look okay?

It does.

> Are there any improvements you'd like to suggest?

None at the moment.

**Reply by pavish** [reply]
> What went well over the past week?

We got the review completed for add records, which was blocking the PR.

> What could have gone better over the past week?

Progress has been slow from my end, due to some unforeseen health concerns. I tried to work through that which delayed progress, it would have been faster and better if I had taken some time off and then worked.

> Does your workload for this week look okay?

It does

> Are there any improvements you'd like to suggest?

None, at the moment.

**Reply by mathemancer** [reply]
> What went well over the past week?

I'm happy to have gotten some types implemented, and to have made good progress on the server default settings.

> What could have gone better over the past week?

I'm trying to take a stance of being quicker to set things down and move to other implementation tasks when something gets blocked or needs clarification, but I think this is resulting in a stack of half-finished things.  I'm currently managing these so they don't get to where the current work on them is unusable, but does add some complexity.  

> Does your workload for this week look okay?

Overall, it's a lot.  If everything goes well, it's feasible, but there's at least one blocker since the `TIME` types probably should be done before the `TIMESTAMP` types are started (since they overlap quite a bit).  

> Are there any improvements you'd like to suggest?

None at the moment.

**Reply by kgodey** [reply]
> What went well over the past week?

I'm glad to have the staging server up finally.

> What could have gone better over the past week?

I haven't had a lot of time to focus on code or design reviews and the new Mathesar project automation is trickier than I thought it would be. I haven't gotten as much done as I'd like.

> Does your workload for this week look okay?

I probably will not be able to get it all done, but that's fine.

> Are there any improvements you'd like to suggest?

No.

**Comment by kgodey** [comment]
## Comms Liaison Schedule

Here's our (randomized) comms liaison schedule for the next few weeks:

| Week | Person |
|-|-|
| Oct 6 to Oct 12 | @ghislaineguerin |
| Oct 13 to Oct 19 | @mathemancer |
| Oct 20 to Oct 26 | @kgodey |
| Oct 27 to Nov 2 | @pavish |

**Comment by kgodey** [comment]
## Active Work

This post lists work planned for this week (see also: [Active](https://github.com/centerofci/mathesar/projects/1)). It also includes a current summary of milestones involved in this week's work.

### [2021-10-08] [[05] Editable Tables](https://github.com/centerofci/mathesar/milestone/4)

- **[9 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%22%5B05%5D%20Editable%20Tables%22) already closed.** 
- **Issues currently in progress:**
  - [@pavish] #197

### [2021-10-13] [[06] 2021-10 Stability](https://github.com/centerofci/mathesar/milestone/34)

- **[7 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%22%5B06%5D%202021-10%20Stability%22) already closed.** 
- **Issues currently in progress:**
  - [@ghislaineguerin] #473
  - [@pavish] #651
  - [@pavish] #639
  - [@kgodey] #280
  - [@kgodey] #592
  - [@bohemia420] #668
  - [@bohemia420] #635
- **Issues not yet started:**
  - [@kgodey] #583
  - [@ghislaineguerin] #636
  - [@ghislaineguerin] #638
  - [@pavish] #640
  - [@pavish] #641
  - [@pavish] #642
  - [@pavish] #619

### [2021-10-15] [[07] Working with Tables](https://github.com/centerofci/mathesar/milestone/8)

- **[22 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%22%5B07%5D%20Working%20with%20Tables%22) already closed.** 
- **Issues currently in progress:**
  - [@mr-gabe49] #448
  - [@dmos62] #498
  - [@mr-gabe49] #496
- **Issues remaining in backlog:**
  - 10 [frontend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20frontend%22+milestone%3A%22%5B07%5D%20Working%20with%20Tables%22) issues

### [2021-10-22] [[08] Initial Data Types](https://github.com/centerofci/mathesar/milestone/11)

- **[33 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%22%5B08%5D%20Initial%20Data%20Types%22) already closed.** 
- **Issues currently in progress:**
  - [@mathemancer] #412
  - [@mathemancer] #682
  - [@mathemancer] #681
  - [@mathemancer] #417
  - [@eito-fis] #426
  - [@mathemancer] #385
  - [@mathemancer] #387
- **Issues not yet started:**
  - [@mathemancer] #384
  - [@mathemancer] #424
  - [@mathemancer] #430
  - [@mathemancer] #420
  - [@mathemancer] #410
  - [@kgodey] #661
  - [@kgodey] #658
- **Issues remaining in backlog:**
  - 16 [backend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20backend%22+milestone%3A%22%5B08%5D%20Initial%20Data%20Types%22) and [database](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20database%22+milestone%3A%22%5B08%5D%20Initial%20Data%20Types%22) issues
  - 9 [frontend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20frontend%22+milestone%3A%22%5B08%5D%20Initial%20Data%20Types%22) issues
  - 8 [product](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20product%22+milestone%3A%22%5B08%5D%20Initial%20Data%20Types%22) issues

### [2021-10-31] [[08A] 2021-10 improvements](https://github.com/centerofci/mathesar/milestone/31)

- **Issues currently in progress:**
  - [@arvind-1] #555
  - [@kgodey] #685
  - [@silentninja] #421
- **Issues not yet started:**
  - [@pavish] #643
- **Issues remaining in backlog:**
  - 6 [backend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20backend%22+milestone%3A%22%5B08A%5D%202021-10%20improvements%22) and [database](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20database%22+milestone%3A%22%5B08A%5D%202021-10%20improvements%22) issues
  - 12 [frontend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20frontend%22+milestone%3A%22%5B08A%5D%202021-10%20improvements%22) issues
  - 3 [infrastructure](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20infrastructure%22+milestone%3A%22%5B08A%5D%202021-10%20improvements%22) issues

### [2021-11-05] [[09] Working with Views](https://github.com/centerofci/mathesar/milestone/13)

- **[4 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%22%5B09%5D%20Working%20with%20Views%22) already closed.** 
- **Issues currently in progress:**
  - [@ghislaineguerin] #455
  - [@ghislaineguerin] #443
  - [@ghislaineguerin] #466
  - [@ghislaineguerin] #454
  - [@ghislaineguerin] #243
  - [@ghislaineguerin] #456
- **Issues remaining in backlog:**
  - 7 [backend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20backend%22+milestone%3A%22%5B09%5D%20Working%20with%20Views%22) and [database](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20database%22+milestone%3A%22%5B09%5D%20Working%20with%20Views%22) issues
  - 2 [product](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20product%22+milestone%3A%22%5B09%5D%20Working%20with%20Views%22) issues

### [2021-11-12] [[10] Data Modeling](https://github.com/centerofci/mathesar/milestone/24)

- **[4 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%22%5B10%5D%20Data%20Modeling%22) already closed.** 
- **Issues currently in progress:**
  - [@rajeshkumarrs] #394
- **Issues remaining in backlog:**
  - 3 [backend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20backend%22+milestone%3A%22%5B10%5D%20Data%20Modeling%22) and [database](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20database%22+milestone%3A%22%5B10%5D%20Data%20Modeling%22) issues
  - 2 [frontend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20frontend%22+milestone%3A%22%5B10%5D%20Data%20Modeling%22) issues
  - 2 [design](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20design%22+milestone%3A%22%5B10%5D%20Data%20Modeling%22) issues

### [2021-11-19] [[11] Multiple Database Support](https://github.com/centerofci/mathesar/milestone/5)

- **[7 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%22%5B11%5D%20Multiple%20Database%20Support%22) already closed.** 
- **Issues currently in progress:**
  - [@aaditkamat] #395
- **Issues remaining in backlog:**
  - 3 [backend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20backend%22+milestone%3A%22%5B11%5D%20Multiple%20Database%20Support%22) and [database](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20database%22+milestone%3A%22%5B11%5D%20Multiple%20Database%20Support%22) issues
  - 3 [frontend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20frontend%22+milestone%3A%22%5B11%5D%20Multiple%20Database%20Support%22) issues
  - 2 [design](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20design%22+milestone%3A%22%5B11%5D%20Multiple%20Database%20Support%22) issues

**Comment by kgodey** [comment]
## Retrospective

This post lists issue updates from 2021-09-28 until now by milestone.

### [[05] Editable Tables](https://github.com/centerofci/mathesar/milestone/4)

- **Issues started, not yet finished:**
  - [@pavish] #197

### [[06] 2021-10 Stability](https://github.com/centerofci/mathesar/milestone/34)

- **Issues closed:**
  - [@kgodey] #687
  - [@mr-gabe49] #646
  - [@kgodey] #225
  - [@pavish] #213
- **Issues started, not yet finished:**
  - [@ghislaineguerin] #473
  - [@pavish] #651
  - [@pavish] #639
  - [@kgodey] #280
  - [@kgodey] #592
  - [@bohemia420] #668
  - [@bohemia420] #635
- **New issues created:**
  - #687, created by @kgodey

### [[07] Working with Tables](https://github.com/centerofci/mathesar/milestone/8)

- **Issues closed:**
  - [@pegnott] #495
- **Issues started, not yet finished:**
  - [@mr-gabe49] #448
  - [@dmos62] #498
  - [@mr-gabe49] #496

### [[08] Initial Data Types](https://github.com/centerofci/mathesar/milestone/11)

- **Issues closed:**
  - [@mathemancer] #407
  - [@mathemancer] #405
  - [@mathemancer] #404
- **Issues started, not yet finished:**
  - [@mathemancer] #412
  - [@mathemancer] #682
  - [@mathemancer] #681
  - [@mathemancer] #417
  - [@eito-fis] #426
  - [@mathemancer] #385
  - [@mathemancer] #387

### [[08A] 2021-10 improvements](https://github.com/centerofci/mathesar/milestone/31)

- **Issues started, not yet finished:**
  - [@arvind-1] #555
  - [@kgodey] #685
  - [@silentninja] #421
- **New issues created:**
  - #689, created by @silentninja
  - #685, created by @kgodey

### [[09] Working with Views](https://github.com/centerofci/mathesar/milestone/13)

- **Issues started, not yet finished:**
  - [@ghislaineguerin] #455
  - [@ghislaineguerin] #443
  - [@ghislaineguerin] #466
  - [@ghislaineguerin] #454
  - [@ghislaineguerin] #243
  - [@ghislaineguerin] #456

### [[10] Data Modeling](https://github.com/centerofci/mathesar/milestone/24)

- **Issues started, not yet finished:**
  - [@rajeshkumarrs] #394

### [[11] Multiple Database Support](https://github.com/centerofci/mathesar/milestone/5)

- **Issues started, not yet finished:**
  - [@aaditkamat] #395

### [[13A] 2021-11 improvements](https://github.com/centerofci/mathesar/milestone/32)

- **New issues created:**
  - #683, created by @pavish
